### ISSUE TITLE
error-record-invalid when running on MacOS Server

### ISSUE THREAD

**Issue by pderby** [EsotericSoftware/dnsmadeeasy#4]
Since Nov 18, 2018, getting error-record-invalid when running on MacOS Server

Submitting HTTPS request to DnsMadeEasy with same data as config file  returns "success"

**Comment by NathanSweet** [comment]
HTTPS shouldn't be required. It works for me using:
```http://cp.dnsmadeeasy.com/servlet/updateip?username=XYZ&password=ABC&id=12234&ip=1.2.3.4```
Does that work (using your data)?

**Comment by pderby** [comment]
http;// and https:// both work fine work fine with my data.  I also used BBEDIT to change the config file line endings to CR, CRLF and LF.  All three variants fail.  My password is 15 characters long, might that be a problem?

**Comment by NathanSweet** [comment]
Password length shouldn't matter. I really can't imagine what would be the problem. If you got rate limited for too many requests, it would probably say so and you'd get the same results in a browser. You said "since", was it working previously and now it doesn't? That would sound like DnsMadeEasy changed something, though it works for me (on Windows but it's a very simple app and should be on the same on OS X).

**Comment by pderby** [comment]
Agree.  I grabbed “realDNS” another option listed on DNSMadeEasy and copy/pasted my config info into their program.  Everything is working fine now.

I asked the support people at DNSMadeEasy if they could let me know what was failing on their end… the password or other data.  If/when they get back to me I’ll let you know.

Thanks so much for being so responsive in the middle of your night!

Paul

**Comment by pderby** [comment]
Is there an easy way for someone who knows nothing about JAVA to see what the program is sending from my Mac to DNSMadeEasy?  Maybe a debug setting that can be turned on when the program is launched that shows in the log what the program is sending to DNSMadeEasy?

**Comment by NathanSweet** [comment]
You could use an HTTP proxy that shows you the requests and responses. Eg, ProxyTrace on Windows. Sorry I don't know about Mac, maybe [this](https://www.charlesproxy.com/). Start Java with your proxy settings like:
```java -Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=8800 -jar dnsmadeeasy.jar```
`127.0.0.1` means localhost (your local computer) and `8800` is the port your proxy tool is running on.

**Comment by pderby** [comment]
Nathan,

The Charles proxy debugger is way cool, thanks for the link.  I installed  Charles and here is what is getting sent from my MacOS server:

GET /servlet/updateip?username=&password=&id=&ip=108.28.6.102 HTTP/1.1

So the config file isn’t getting read by and passed along to DNSMadeEasy.

The config file is in

 ~/.dnsmadeeasy/

To start up dnsmadeeasy.jar

sudo java -Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=8800 -jar /Users/admin/Documents/DDNS/dnsmadeeasy.jar

File permission on the config file are

-rw-r--r--@  1 admin  staff   118 Nov 20 12:26 config.txt

Beats me why the java program quit reading the config file.  Maybe Apple did something in a security settings release to block executing JAVA programs from reading files?  I have no idea.  I couldn’t find anything where MACOS SERVER had been changed.

Paul

**Comment by NathanSweet** [comment]
If you delete the config file, then run the program, does it create a new config file? That is one way to be sure the config file it is reading is the one you have edited with your info. When it saves the config file, it will print `Saving: /path/to/file/config.txt`. It *should* be `~/.dnsmadeeasy/config.txt`, but who knows. Oh, maybe using `sudo` runs it as the root user, so it's in the root user's home directory. You can set the folder:
```
java -Duser.home=/some/path -jar dnsmadeeasy.jar
```
That would cause it to use `/some/path/.dnsmadeeasy/config.txt`.

It reading a config file at a different path seems like the most likely problem. Newlines can be a line feed (`\n`), a carriage return (`\r`), or a carriage return followed immediately by a linefeed (`\r\n`).
